tool/lib/commands/update_flutter_sdk.dart

```diff
@@ -133,35 +134,28 @@ class UpdateFlutterSdkCommand extends Command {
       repo.toolDirectoryPath,
       flutterSdkDirName,
     );
-    final toolFlutterSdk = Directory.fromUri(Uri.file(toolSdkPath));
+    final toolFlutterSdk = Directory(toolSdkPath);
     log.stdout('Updating "$toolSdkPath" to branch $flutterTag');
 
     if (toolFlutterSdk.existsSync()) {
-      log.stdout('"$toolSdkPath" directory already exists');
-      await processManager.runAll(
-        commands: [
-          CliCommand.git('fetch'),
-          CliCommand.git('checkout $flutterTag -f'),
-          CliCommand.flutter('--version'),
-        ],
-        workingDirectory: toolFlutterSdk.path,
-      );
-    } else {
-      log.stdout('"$toolSdkPath" directory does not exist - cloning it now');
-      await processManager.runProcess(
-        CliCommand.git(
-          'clone https://github.com/flutter/flutter $flutterSdkDirName',
-        ),
-        workingDirectory: repo.toolDirectoryPath,
-      );
-      await processManager.runAll(
-        commands: [
-          CliCommand.git('checkout $flutterTag -f'),
-          CliCommand.flutter('--version'),
-        ],
-        workingDirectory: toolFlutterSdk.path,
-      );
+      toolFlutterSdk.deleteSync(recursive: true);
```

Comment:
What's the reason for deleting and re-cloning this? (seems like that would be way slower than the original code that just pulled latest commits)

Reply:
I kept getting process exceptions on the series of `git` calls, so figured deleting and re-cloning would be a robust way to make sure we get the branch we need and aren't getting in a bad state from git.

Reply:
Reverted back to what we had. If I see the errors again we can dig back into this issue. maybe I got myself in a weird state by cancelling the script in the middle of execution.